src/Tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema.UnitTests/SchemaGenerationTests.cs

```diff
@@ -0,0 +1,107 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using Xunit.Abstractions;
+using NJsonSchema;
+using Microsoft.Diagnostics.Monitoring.UnitTests.Options;
+using System.IO;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System;
+using System.Reflection;
+using NJsonSchema.Generation;
+
+namespace Microsoft.Diagnostics.Monitoring.ConfigurationSchema.UnitTests
+{
+    public class SchemaGenerationTests
+    {
+        private readonly ITestOutputHelper _outputHelper;
+
+        private const string SchemaBaseline = "schema.json";
+
+        private static readonly string CurrentExecutingAssemblyPath =
+            Assembly.GetExecutingAssembly().Location;
+
+        private static readonly string SchemaBaselinePath =
+            Path.Combine(Path.GetDirectoryName(CurrentExecutingAssemblyPath), SchemaBaseline);
+
+        public SchemaGenerationTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+        }
+
+        [Fact]
+        public void TestGenerateSchema()
+        {
+            string completeSchema = GenerateSchema();
+            _outputHelper.WriteLine(completeSchema);
+        }
+
+        [Fact]
+        public async Task TestSchemaBaseline()
+        {
+            string generatedSchema = GenerateSchema();
+            using StreamReader baseline = new StreamReader(SchemaBaselinePath);
+            using StringReader generated = new StringReader(generatedSchema);
+            bool equal = await CompareLines(baseline, generated);
+            Assert.True(equal, "Generated schema differs from baseline.");
+        }
+
+        private string GenerateSchema()
```

Comment:
Could this be moved to its own executable so that it's easier to accept the output of the generation as the new baseline?